// beautylog/dist/beautylog.classes.ora.js
"use strict";
require("typings-global");
const plugins = require("./beautylog.plugins");
const beautylog_log_helpers_1 = require("./beautylog.log.helpers");
exports.oraActive = false; // when an Ora is active (e.g. start()) this is true;
class Ora {
    constructor(textArg, colorArg, startArg = false) {
        this._oraObject = plugins.ora({
            spinner: "dots",
            text: textArg,
            color: colorArg
        });
        startArg ? this._oraObject.start() : void (0);
    }
    text(textArg) {
        this._oraObject.text = textArg;
    }
    ;
    start(textArg, colorArg) {
        if (textArg)
            this._oraObject.text = textArg;
        if (colorArg)
            this._oraObject.color = colorArg;
        exports.activeOra = this;
        exports.oraActive = true;
        this._oraObject.start();
    }
    ;
    end() {
        this._oraObject.stop();
        this._oraObject.clear();
        exports.activeOra = undefined;
        exports.oraActive = false;
    }
    endOk(textArg) {
        this.end();
        beautylog_log_helpers_1.logNode("ok", textArg);
    }
    ;
    endError(textArg) {
        this.end();
        beautylog_log_helpers_1.logNode("error", textArg);
    }
    ;
    pause() {
        this._oraObject.stop();
    }
    stop() {
        this.end();
    }
    ;
}
exports.Ora = Ora;
